### PR TITLE
update code of conduct charter to address other committee elections

### PR DESCRIPTION
**Pull request by salaxander**
Over the course of the code of conduct committee's existence, we have seen a number of members run for steering while serving on the committee. In the event that they are elected, they resign from code of conduct (serving on both would be a conflict of interest). This does however create the potential for high turnover on code of conduct if many members end up making that a stepping stone to steering.

I am proposing this charter change to encourage elected members to remain with code of conduct through their committed term.

### CODE REVIEW

File: committee-code-of-conduct/charter.md
Comment (detiber) [inline committee-code-of-conduct/charter.md:154]: To align with the above paragraph about resignation, would it be better to say that they are required to announce their resignation from the CoCC with the resignation taking effect the date the election ends? That would allow for a notice period and at least a short overlap with their replacement for continuity purposes.

I think it would also be good to call out specifically that this applies only to committees that would cause a conflict of interest, since I don't think such a conflict of interest is present with the Security Response Committee.

Comment (salaxander) [reply committee-code-of-conduct/charter.md]: Yeah maybe at this time it makes sense to keep it specific to steering

**Review by endocrimes — APPROVED** [review]
/lgtm

This should greatly simplify the life-cycling and continuity of the CoCC - especially given that both elections often happen around the same time.

**Comment by cblecker** [comment]
/hold
/cc @kubernetes/steering-committee 
/label committee/steering

Explicit hold as all charter changes require steering review

**Comment by cblecker** [comment]
/committee steering

Let's try that again

**Comment by tpepper** [comment]
Compared to stated intent in the PR description, I'm not sure I see in the charter change a strong incentivize for that outcome.  Pondering if/how we might more strongly....

**Comment by endocrimes** [comment]
@tpepper I don't think we want to _stop_ people from doing so if that's what they desire, but want to make it less of a... "tossing the hat in the ring and then making a decision" (which is fairly disruptive to both of our election processes), but more "committing to staying on the CoCC or running for steering".

File: committee-code-of-conduct/charter.md
Comment (BenTheElder) [inline committee-code-of-conduct/charter.md:156]: just to clarify: "on the date of the election", can we be extra-specific ... the date on which the election concludes or the date on which the voting begins? (I'd guess the former, to create a transition period leading up to potentially being on steering, but I'm not actually certain if that's the intent).

**Review by BenTheElder — COMMENTED** [review]
Restating to make sure I'm reading / understanding this correctly, please correct me if not, brain is fried at the moment 🙃 

- This would effectively give a month of guaranteed transition time, should someone on CoCC opt to run for steering.
  - This gives CoCC time to prepare and transition in a new member smoothly, as opposed the immediate step down when a member suddenly wins steering today.

- It further gives pause to casually running for steering in the middle of a CoCC term, because regardless of winning you will be stepping down from CoCC. 
  - This is important because CoCC members have been known to win (but could lose), and the unpredictable turnover disrupts the intended staggered terms.

This seems very reasonable, and IIRC we already do not allow the opposite transition from Steering directly to CoCC.

I have one clarification ask regarding which election date we mean in the charter text.

**Comment by BenTheElder** [comment]
To the discussion above, I might suggest updating the PR body re:

> I am proposing this charter change to encourage elected members to remain with code of conduct through their committed term.

To also note the positive effect of ensuring a predictable transition period if members do go forward with running anyhow.

Just for anyone coming across this later.

**Comment by salaxander** [comment]
> /hold /cc @kubernetes/steering-committee /label committee/steering
> 
> Explicit hold as all charter changes require steering review

I may want to add this statement to the charter as part of this too then. Currently it states "Any changes to the charter require explicit LGTM or Approve from all committee members. For pull requests, a /hold will be applied until all approvals are present. Any changes merged without consensus will be reverted". I think it makes total sense to have approval from steering as well, I just want to be sure that requirement is documented.

File: committee-code-of-conduct/charter.md
Comment (salaxander) [reply committee-code-of-conduct/charter.md:156]: Good catch! I'll clarify this

**Comment by cblecker** [comment]
@salaxander Yes, it might make sense to call that out. You could link it to https://github.com/kubernetes/community/tree/master/committee-steering/governance#sig-charter-approval-process which is the relevant section in the governance docs

File: committee-code-of-conduct/charter.md
Comment (detiber) [inline committee-code-of-conduct/charter.md:152]: This is kind of nitpicky, but "run for election for" reads a bit off to me.

Maybe "If a current code of conduct committee member chooses to participate in the Kubernetes steering committee election"

or "If a current code of conduct committee member chooses to run for the Kubernetes steering committee"

or possibly even "If a current code of conduct committee member chooses to accept nomination for a Kubernetes steering committee election and becomes a qualified candidate, they are required to...".

In the case of the later, it would probably also be good to also specify the definitions of accepting nomination and becoming a qualified candidate come from https://github.com/kubernetes/steering/blob/main/elections.md and https://github.com/kubernetes/community/tree/master/elections/steering

**Comment by jberkus** [comment]
Counter-argument:  recruiting folks for CoCC *at all* is very difficult, and the pool of potential CoCC members has like 70% overlap with the pool of potential SC members.  Are we sure that we want to position things so that potential candidates say "hmmm, I don't want to run for CoCC because I might want to run for SC next year"?

**Comment by salaxander** [comment]
> Counter-argument: recruiting folks for CoCC _at all_ is very difficult, and the pool of potential CoCC members has like 70% overlap with the pool of potential SC members. Are we sure that we want to position things so that potential candidates say "hmmm, I don't want to run for CoCC because I might want to run for SC next year"?

I think that's a valid concern. Maybe we as the current committee need to do a better job of sourcing candidates for future roles? As it is though, the high turnover and lack of continuity ends up being quite disruptive to the code of conduct committee. I'm open to exploring other options, but we really would like to find a way to ensure that we don't find ourselves in a situation like the current one again (every member of the committee is new)

**Comment by tpepper** [comment]
The thing I’ve been trying to figure out: how do we incentivize finishing out terms?  Admittedly I didn’t on CoCC myself. And while I feel like I need to go back and deliver on that commitment, it wont be easy: to stand for next year’s 2023 CoCC election I’d have to step down early from SC. Or I would need to not stand for 2023 SC reelection and and then wait ten months for 2024 CoCC (re)election.  That wouldn’t be helped by first stepping off CoCC per the idea here, and like Josh notes it looses one or both bodies a volunteer. It might only be helped by us individually realizing we need to be finish commitments before aspiring to or taking on next ones. That’s hard to incentivize, especially constructively vs punitively.

Still feels punitive, but would it be more effective to say committee candidacy does not include ones who did not complete a prior committee commitment in the (~1year?) lead up to the committee election?

**Comment by cblecker** [comment]
Top of head thought: would it make this problem better/worse/no change if we aligned the election cycles between the committees

**Comment by jberkus** [comment]
> Top of head thought: would it make this problem better/worse/no change if we aligned the election cycles between the committees

Currently, the SC administers the CoC elections.  So that would shift to having the "lame duck" SC administer them.  Which isn't necessarily a blocker, just extra overhead.

File: committee-code-of-conduct/charter.md
Comment (Chiefy0x1) [inline committee-code-of-conduct/charter.md:98]: Thanks

**Review by Chiefy0x1 — APPROVED** [review]
/approve 
LGTM it reads as do the next right thing. I like that..

File: committee-code-of-conduct/charter.md
Comment (mrbobbytables) [inline committee-code-of-conduct/charter.md:159]: Can we break this out into a separate PR? I don't think theres any debate on this part of it and we should ideally get it merged before the actual election kicks off :x

**Review by Chiefy0x1 — APPROVED** [review]
/LGTM

**Pull request closed without merging**
